src/orient/rational-calc.orient

```diff
@@ -0,0 +1,273 @@
+        describe(post_mtree_cached_hashnodes, "Number of intermediary hash nodes cached of the replica merkle tree")
+        describe(post_mtree_cached_gb, "POST_MTREE_CACHED in GiB")
+        describe(post_mtree_overhead, "Fraction of extra storage a miner will have to use to cache the replica merkle tree")
+        describe(post_mtree_layers_cached, "Number of layers of the replica merkle tree being cached")
+        describe(post_mtree_layers_deleted, "Number of layers of the replica merkle tree not being cached")
+
+        post_mtree_cached = post_mtree_cached_hashnodes * node_size
+        post_mtree_cached_gb = post_mtree_cached / (1024 * 1024 * 1024)
+        post_mtree_overhead = post_mtree_cached / sector_size
+        post_mtree_cached_hashnodes = expt(2, post_mtree_layers_cached) - 1
+        post_mtree_layers_deleted = commr_tree_depth - post_mtree_layers_cached
+
+      Time:
+        // Note: this assumes reading the cached parts of the merkle tree instantaneous
+        describe(post_witness_gen, "Time to regenerate and to read the merkle tree inclusion proofs for the challenged nodes from memory")
+
+        post_witness_gen = post_mtree_regen_challenges_time + post_mtree_access_challenges_time
+
+        Regeneration:
+          describe(post_mtree_regen_hashnodes, "Number of intermediary nodes to be regenerated for a single replica merkle tree inclusion proof in PoSt")
+          describe(post_mtree_regen_challenge_time, "Time to generate the missing intermediary hash nodes of the replica merkle tree inclusion proof for a single challenge in PoSt")
+          describe(post_mtree_regen_challenges_time, "Same as post_mtree_regen_challenge_time but for all the PoSt challenges")
+
+          post_mtree_regen_hashnodes = expt(2, post_mtree_layers_deleted + 1) - 1
+          post_mtree_regen_challenge_time = (post_mtree_regen_hashnodes * 2) * merkle_tree_hash_time
+          post_mtree_regen_challenges_time = post_mtree_regen_challenge_time * post_challenges
+
+        Access:
+          describe(post_mtree_access_challenge_time, "Time to sequentially read the replica leaves needed to regenerate the missing layers of the merkle tree for a single challenged node in PoSt")
+          describe(post_mtree_access_challenges_time, "Same as post_mtree_access_challenge_time but for all the PoSt challenges")
+
+          // TODO: this is using random access, we should use seq access instead
+          post_mtree_access_challenge_time = post_mtree_regen_hashnodes * rig_storage_latency
+          post_mtree_access_challenges_time = post_mtree_access_challenge_time * post_challenges
+          // TODO: log2(nodes) > post_mtree_layers_cached
+
+    ProofGeneration:
+      describe(post_proof_gen, "Time to generate a PoSt Proof (including SNARKs, generating witnesses and reading from storage the challenged leaves)")
+
+      post_proof_gen = post_snark_time + post_witness_gen 
```

Comment:
`post_snark_time` does not seem to be defined anywhere.

Reply:
yes, I will try to have some form of "time cost per constraint"